react/src/pages/AntMedia.js

```diff
@@ -463,73 +464,119 @@ function AntMedia(props) {
         } else if (info === "publish_started") {
             speedTestCounter.current = 0;
             console.log("speed test publish started");
-            let tempSpeedTestObject = {};
-            tempSpeedTestObject.message = speedTestObject.message;
-            tempSpeedTestObject.isfinished = false;
-            tempSpeedTestObject.isfailed = false;
-            tempSpeedTestObject.errorMessage = "";
-            tempSpeedTestObject.progressValue = 20;
-            setSpeedTestObject(tempSpeedTestObject);
+            setSpeedTestObjectProgress(20);
             speedTestForPublishWebRtcAdaptor.current.enableStats("speedTestStream" + speedTestStreamId.current);
         } else if (info === "updated_stats") {
-            speedTestCounter.current = speedTestCounter.current + 1;
-
-            let tempSpeedTestObject = {};
-            tempSpeedTestObject.message = speedTestObject.message;
-            tempSpeedTestObject.isfinished = false;
-            tempSpeedTestObject.isfailed = false;
-            tempSpeedTestObject.errorMessage = "";
-            tempSpeedTestObject.progressValue = 20 + (speedTestCounter.current * 30);
-            setSpeedTestObject(tempSpeedTestObject);
-
-            if (speedTestCounter.current > 2) {
-                speedTestForPublishWebRtcAdaptor.current?.stop("speedTestStream" + speedTestStreamId.current);
-                let rtt = ((parseFloat(obj.videoRoundTripTime) + parseFloat(obj.audioRoundTripTime)) / 2).toPrecision(3);
-                let packetLost = parseInt(obj.videoPacketsLost) + parseInt(obj.audioPacketsLost);
-                let jitter = ((parseFloat(obj.videoJitter) + parseInt(obj.audioJitter)) / 2).toPrecision(3);
-                let outgoingBitrate = parseInt(obj.currentOutgoingBitrate);
-                let bandwidth = parseInt(speedTestForPublishWebRtcAdaptor.current.mediaManager.bandwidth);
-                console.log("* rtt: " + rtt);
-                console.log("* packetLost: " + packetLost);
-                console.log("* jitter: " + jitter);
-                console.log("* outgoingBitrate: " + outgoingBitrate);
-                console.log("* bandwidth: " + bandwidth);
-
-                let speedTestResult = {};
-
-                if (rtt >= 150 || packetLost >= 2.5 || jitter >= 80 || ((outgoingBitrate / 100) * 80) >= bandwidth) {
-                    console.log("-> Your connection quality is poor. You may experience interruptions");
-                    speedTestResult.message = "Your connection quality is poor. You may experience interruptions";
-                } else if (rtt >= 50 || packetLost >= 1 || jitter >= 30 || outgoingBitrate >= bandwidth) {
-                    console.log("-> Your connection is fair, but you may experience interruptions");
-                    speedTestResult.message = "Your connection is fair, but you may experience interruptions";
-                } else {
```

Comment:
Add an extra level rtt >= 50 || packetLostPercentage >= 1 || jitter >= 30 with a different message 

Your connection is good!